### ISSUE TITLE
Blocks use custom selection color

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15270]
Blocks use custom selection color

-   **Severity:** Low
-   **Affected Populations:**
    -   Low-Vision
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Block Editing
    -   Block Options

#### Issue description

Blocks use a custom CSS color for text selection.

For Firefox users who have low-vision and use a Windows high-contrast
theme, custom selection colors are not honoured, but are not replaced by
defaults either. Consequently, these users will not be able to see what
text is selected.

##### Issue Code

```
    .editor-block-list__layout .editor-block-list__block ::selection {
        background-color: #b3e7fe;
    }
```

#### Remediation Guidance

Remove custom selection rules, and allow the browser/OS native color to
be rendered.

#### Relevant standards

-   [1.3.3 Sensory Characteristics
    (Level A)](https://www.w3.org/TR/WCAG20/#content-structure-separation-understanding)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-97 in Tenon's report

**Comment by afercia** [comment]
To my knowledge, Firefox honors `::selection` (at least on macOS?). I'd agree on the general issue that the best option is to not play with operating system defaults. Will take care to propose this issue for discussion during next accessibility team meeting.

**Comment by melchoyce** [comment]
Full report screenshot:

![image](https://user-images.githubusercontent.com/2846578/57186039-f7009c80-6e8c-11e9-9dd1-0e60d23c701d.png)

**Comment by StommePoes** [comment]
> To my knowledge, Firefox honors ::selection (at least on macOS?). 
MacOS doesn't have a real full high contrast setting (to my knowledge, unless I'm out of date).
I could have sworn there was a filed Bugzilla on this however I am having trouble finding it. :(
It is still easily reproduceable on a fresh latest Firefox and Windows High Contrast however:
![Four cropped screenshots of Firefox: first is an invisible selection in windows high contrast; second is the selection after focus has left Firefox; third is the selection without windows high contrast; last is the selection after focus has left Firefox](https://user-images.githubusercontent.com/136210/57194616-24128500-6f49-11e9-87cd-23c8e7a5c021.png)

The second and last images are when you have selected something and then for example clicked another desktop application (allowing focus to leave Firefox without removing the selection).

These are without any changes to Firefox colour settings, which I kind of assume would change these results? But I'm not sure.

**Comment by afercia** [comment]
Right, I missed the "and use a Windows high-contrast theme" part :)

**Comment by mapk** [comment]
I'll take a look at this. In [today's triage on slack](https://wordpress.slack.com/archives/C02S78ZAL/p1557850452309000), the Design Team agreed to just use the browser's default highlight if possible. How this effects high contrast modes will probably need further work.

**Comment by StommePoes** [comment]
Leaving selection to the default will Just Work.
This bug is something to do with how Firefox deals with how it reads the Highlight colour from Windowing software (so Windows just has... Windows) as there are bugs on similar issues on Linux (where various Gnome High Contrast schemes meet terrible ends when run with Firefox which also allows users to set certain window colours, as well as how FF deals with windowing themes... it's a bit of a mess).

**Comment by mapk** [comment]
So @StommePoes, thanks for your detailed comments. In your experience, Windows high contrast mode in FF will still show the text as highlighted if we use the default browser selection highlight? Is that correct?

**Comment by m-e-h** [comment]
This issue was also brought up in sanitize.css. https://github.com/csstools/sanitize.css/issues/173

**Comment by StommePoes** [comment]
@mapk correct.
